src/lib/services/segment-service.ts

```diff
@@ -82,17 +96,25 @@ export class SegmentService implements ISegmentService {
 
         await this.eventStore.store({
             type: SEGMENT_CREATED,
-            createdBy: user.email || user.username,
+            createdBy: user.email || user.username || 'unknown',
             data: segment,
         });
 
         return segment;
     }
 
-    async update(
+    async update(id: number, data: unknown, user: User): Promise<void> {
+        if (this.flagResolver.isEnabled('segmentChangeRequests')) {
```

Comment:
we have to be careful not to enable it before the UI is ready

Reply:
Fair. It's currently enabled in dev, but we can turn that off